### PR TITLE
[trainer] round numbers in trainer state

### LINKED ISSUE #9475: [trainer] fractional epoch
Running 
```
export BS=20; rm -r output_dir; CUDA_VISIBLE_DEVICES=0,1 PYTHONPATH=../../src USE_TF=0   ./finetune_trainer.py --model_name_or_path t5-small --output_dir output_dir --adam_eps 1e-06 --data_dir wmt_en_ro --do_eval --do_predict --do_train --evaluation_strategy=steps --freeze_embeds --label_smoothing 0.1 --learning_rate 3e-5 --logging_first_step --logging_steps 1000 --max_source_length 128 --max_target_length 128 --num_train_epochs 1 --overwrite_output_dir --per_device_eval_batch_size $BS --per_device_train_batch_size $BS --predict_with_generate --eval_steps 25000  --sortish_sampler --task translation_en_to_ro --test_max_target_length 128 --val_max_target_length 128 --warmup_steps 5 --n_train 100 --n_val 100 --n_test 100  --fp16 --save_steps 1
```
on master, gives:
```
{'loss': 14.846837043762207, 'learning_rate': 6e-06, 'epoch': 0.3333333333333333} 
```
epoch can't be fractional.

### PR DESCRIPTION
**Pull request by stas00**
This PR rounds very long fractions in trainer state e.g., 

```
{'loss': 14.846837043762207, 'learning_rate': 6e-06, 'epoch': 0.3333333333333333} 
```
to:
* epoch 2 decimals
* loss 4 decimals

resulting in:

```
{'loss': 14.8468, 'learning_rate': 6e-06, 'epoch': 0.33}
```` 

If you want any other small tweaks for me to add please let me know.

Fixes: #9475 

@sgugger

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
LGTM!

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -896,7 +896,7 @@ def train(self, model_path: Optional[str] = None, trial: Union["optuna.Trial", D
                     self.lr_scheduler.step()
                     model.zero_grad()
                     self.state.global_step += 1
-                    self.state.epoch = epoch + (step + 1) / steps_in_epoch
+                    self.state.epoch = round(epoch + (step + 1) / steps_in_epoch, 2)
```
Comment (sgugger) [inline src/transformers/trainer.py:899]: No the value should only be rounded when logged. Internal code could use `self.state.epoch` and needs the exact value.

Comment (stas00) [reply src/transformers/trainer.py]: You are correct, logging time it is. Done
